packages/api/src/hooks/p2p/useAdvertInfo.ts

```diff
@@ -0,0 +1,45 @@
+import { useMemo } from 'react';
+import useQuery from '../../useQuery';
+
+/**
+ * This custom hook returns the advert info for a specific advert by calling 'p2p_advert_info' endpoint
+ */
+const useAdvertInfo = (
+    payload: NonNullable<Parameters<typeof useQuery<'p2p_advert_info'>>[1]>['payload'],
+    options?: NonNullable<Parameters<typeof useQuery<'p2p_advert_info'>>[1]>['options']
+) => {
+    const { data, ...rest } = useQuery('p2p_advert_info', {
+        payload: { id: payload?.id, use_client_limits: payload?.use_client_limits },
+        options,
+    });
+
+    const modified_data = useMemo(() => {
+        const p2p_advert_info = data?.p2p_advert_info;
+
+        if (!p2p_advert_info) return undefined;
+
+        return {
+            ...p2p_advert_info,
+            /** Determines whether the advert is a buy advert or not. */
+            is_buy: p2p_advert_info?.type === 'buy',
+            /** Determines whether the advert is a sell advert or not. */
+            is_sell: p2p_advert_info?.type === 'sell',
+            is_block_trade: Boolean(p2p_advert_info?.block_trade),
```

Comment:
I guess you don't need these optional chains cause you are checking if data is not undefined then return this obj